### ISSUE TITLE
No Course assignment if clicked on course link instead of registration v 1.11.16

### ISSUE THREAD

**Issue by Bavarianspirit** [chamilo/chamilo-lms#4147]
**Describe**
On Courselist if someone click on the Course link instead of Register. Course can be done but there ist no Course registration
you can see no subscription to the course and no results are shown.

**To Reproduce**
Log in as normal user!
1. Go to Course List
2. Click on the Text Link (blue Course Link) in Course overview instead of Register (green button)
3. Now you have to accept the use terms of this course
4. Log in as Admin
5. go to admin Panel and look at the user. No subscription is done.

You can do the subscription manually. assign the use to the course. Now you can see the quiz results.

![grafik](https://user-images.githubusercontent.com/35309163/151261089-248cb15c-7483-4d6e-8bfd-ccd8f5fceb42.png)

![grafik](https://user-images.githubusercontent.com/35309163/151262192-eb4042ff-d7c5-4b8e-985e-b53344c068fd.png)
if you click on Testkurs there will be no Course assignment
if you cklick on Registrieren the assignment will be correct.

I tried this on some chamilo installations. from 1.10.x to 1.11.x always the same

**Comment by Bavarianspirit** [comment]
For futher info:
The Link Testkurs is the same as u click on got to course (the symbol instead registration - if you are registered already)
https://URL/courses/tk001/index.php?id_session=0

The Link for Registration is:
https://URL/main/auth/courses.php?action=subscribe_course&sec_token=839f696c98591297bd7128a3f2a8e5f3&course_code=tk001&search_term=&category_code=

if you are not registered and you click on the 1. Link
https://URL/courses/tk001/index.php?id_session=0
you can do this course but u are not registered

If the Course Admin will look at:
https://URL/main/admin/user_information.php?user_id=24

there are no informations about course progress. Because there is no registration. If the admin do the registration manually, then the progress is visible. Also the progress before manual registration by the admin.

Also if the user click on Testcourse, the terms will be displayed, he can agree and can do the course. But if he will return to my courses, it will not be displayed. Also on Course list, the Registration button is still there and there is no  go to course button.

Can someone fix this issue.
Or is it possible to paste the fix here for 1.11.16

Thanks

**Comment by ywarnier** [comment]
@Bavarianspirit What visibility did you give to your course?

**Comment by Coursenligne** [comment]
I just tested using fresh install of 1.11.16.
I set the objectives and course info.
Whether I click on the link or "Register", Chamilo displays the same page wher I have to register in order to attend the course.
The course is open ; auto-registration is allowed (for this specific course) and unregistering is not allowed.

**Comment by NicoDucou** [comment]
This is correct that when accessing an open course directly you do not get registered to the course and do not see it on the My course page because this correspond to the use case like this.
if you need the user to be registered in the course then the best option would be to change the destination of the link in the catalog with the following option in app/config/configuration.php (this option is available since the 31st of march 2020) :

```
$_configuration['course_catalog_settings'] = [
    'link_settings' => [
        'info_url' => 'course_description_popup', // course description popup page
        'title_url' => 'course_home', // Course home URL
        'image_url' => 'course_about', // Course about URL
    ]
]
```
So the user will need to click on register to be able to access the course. 
Once the user is subscribed to the course he can see the "Go to the course" button in the catalog.

Just to remember the commit referenced in this issue does not correspond to the description and is not included in Chamilo.

@Bavarianspirit could you confirm it is good for you with this settings so that you can force the user to subscribe to the course if it is necessary in your use case.

Best regards

**Comment by Bavarianspirit** [comment]
i use this fix some month and it works without failure:
https://github.com/chamilo/chamilo-lms/pull/4270/commits/71fb6ab2757e4c065063a5531e63e8f937ab4ea4
there is nothing to do.
You can close this issue and pull this fix into the next release:
https://github.com/chamilo/chamilo-lms/pull/4270/commits/71fb6ab2757e4c065063a5531e63e8f937ab4ea4

**Comment by Bavarianspirit** [comment]
This Fix works also in 1.11.28 please merge it.
